### PR TITLE
Introduce git submodule dev-tools

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
Removed `tools` directory and added git submodule `dev-tools`.
You can treat it as a symlink to the remote repo which contains
all developer's files for working.

### CODE REVIEW

**Comment by ArthurBandaryk** [comment]
There is already a [PR](https://github.com/3rdparty/eventuals/pull/141) which fixes Code Style Check failure (there is a line which exceeds 80 chars).

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
+.vscode
+.DS_Store
 bazel-bin
 bazel-out
 bazel-eventuals
-bazel-testlogs
\ No newline at end of file
+bazel-testlogs
+bazel-stout-eventuals
```
Comment (benh) [inline .gitignore:7]: That's an old directory that you can delete on your local machine and should not be generated again in the future!

File: .github/workflows/check_format_code.yml [hunk 2]
```diff
@@ -7,16 +7,7 @@ on:
...
-    steps:
-    - uses: actions/checkout@v2
-    - name: Run bootstrap
-      run: cd tools/ && ./bootstrap
-    - name: Run clang-format style check for C/C++ programs.
-      uses: jidicula/clang-format-action@v4.0.0
-      with:
-        clang-format-version: '12'
-        check-path: '.'
+  check_code_format:
```
Comment (benh) [inline .github/workflows/check_format_code.yml:12]: Minor nit but let's fix it please: `check_code_format` but the file is called `check_format_code`, can we make everything be `check_code_format` please? Thanks!

**Review by benh — APPROVED** [review]
Please address comments and ship it!

File: .github/workflows/check_format_code.yml (hunk 2 above)
Comment (ArthurBandaryk) [reply .github/workflows/check_format_code.yml:12]: Sure! Great catch. I will make it be `check_code_format` in `dev-tools` too.

**Pull request merged**